judge/judgedaemon.main.php

```diff
@@ -486,7 +486,6 @@ function fetch_executable_internal(
 }
 
 $options = getopt("dv:n:hVe:j:t:", ["diskspace-error"]);
-// FIXME: getopt doesn't return FALSE on parse failure as documented!
```

Comment:
Where does it say that this bug has been fixed? Would be good to either have an explicit link to the bugfix or have tested it.

Reply:
If you have a way to reproduce this? I tested some bogus invocations and didn't get a failure.

Reply:
That's exactly reproducing problem. If you can the line
```
echo "Error: parsing options failed.\n";
```
triggered, then this comment can be removed.

Reply:
Ok, I'll play around with it a bit more but have removed the commit for now. Let's discuss next weekend.

Reply:
So in normal cases you'll not get `false`, see:
https://bugs.php.net/bug.php?id=81352 (we run PHP>8.0 so will not encounter this anymore)

You might be able to still get it by not registering $argv (https://stackoverflow.com/a/59785550) or using `unset($argv)` which was enough to encounter this.

Also see the note:
```
The parsing of options will end at the first non-option found, anything that follows is discarded.
```
So it passes a string and if it finds anything which is not an option it will stop. So if we would want to handle this perfectly we would need to count the length of the argument $argv and the index where parsing stopped and compare that? I think this would only complicate the code.

Reply:
This shows that `getopt` is buggy:
```
#!/usr/bin/env php
<?php declare(strict_types=1);

$options = getopt("dv:n:hVe:j:t:", ["diskspace-error"], $index);
// FIXME: getopt doesn't return FALSE on parse failure as documented!
if ($options===false) {
    echo "Error: parsing options failed.\n";
    usage();
}
var_dump($index, $options, $argv);
```

Reply:
I agree this is buggy. I don't think there is really more testing we can do apart from writing our own implementation.

Reply:
I've added a wrapper which handles most (if not all?) of the buggy behaviours of the `getopt` function.